mlip/2021_06_09_vertex_ai_predictions/deploy_model_vertexai.ipynb

```diff
@@ -0,0 +1,502 @@
+    "Models to be served on an endpoint need to first be created on [Vertex AI Models](https://console.cloud.google.com/vertex-ai/models). They can be created from a training or by importing from\n",
+    "your model artifacts. In the case of a Tensorflow model, this is the folder containing your variables and .pb file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# TODO: uncomment and fill in\n",
+    "# ARTIFACT_LOCATION_GCS = \"<GCS_PATH>\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Machine type for serving. If required, GPUs can be added. See [guidelines](https://cloud.google.com/vertex-ai/docs/predictions/configure-compute#gpus) which GPUs are supported by the machines."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SERVING_MACHINE_TYPE=\"n1-standard-2\"\n",
+    "SERVING_GPU, SERVING_NGPU = (None, None)  # example: (aip.gapic.AcceleratorType.NVIDIA_TESLA_K80.name, 2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Serving from Vertex AI endpoints is done from Docker images that run an HTTP server. The image is selected here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TF = \"2-4\"  # 1.15 to 2.4 is supported at the time of writing (26/05/2021) \n",
+    "if TF[0] == \"2\":\n",
+    "    if SERVING_GPU:\n",
+    "        SERVING_VERSION = \"tf2-gpu.{}\".format(TF)\n",
+    "    else:\n",
+    "        SERVING_VERSION = \"tf2-cpu.{}\".format(TF)\n",
+    "else:\n",
+    "    if SERVING_GPU:\n",
+    "        SERVING_VERSION = \"tf-gpu.{}\".format(TF)\n",
+    "    else:\n",
+    "        SERVING_VERSION = \"tf-cpu.{}\".format(TF)\n",
+    "\n",
+    "SERVING_IMAGE_URI = \"europe-docker.pkg.dev/vertex-ai/prediction/{}:latest\".format(SERVING_VERSION)\n",
+    "\n",
+    "print(\"Deployment:\", SERVING_IMAGE_URI, SERVING_GPU, SERVING_NGPU)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime\n",
+    "\n",
+    "TIMESTAMP = datetime.now().strftime(\"%Y%m%d%H%M%S\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from google.cloud import aiplatform as aip\n",
+    "from typing import Optional, Sequence, Dict, Tuple"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# API service endpoint\n",
+    "API_ENDPOINT = \"{}-aiplatform.googleapis.com\".format(REGION)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def upload_model_sample(\n",
```

Comment:
Think this sample function is directly from a ai platform sample provided by Google. Shouldn't we mention that somewhere?